beta_product/views.py

```diff
@@ -42,3 +41,46 @@ def post(self, request):
         )
         serializer = UserInterestSerializer(user_interest)
         return Response(serializer.data, status=status.HTTP_201_CREATED)
+
+
+class UserInterestValidApiView(APIView):
+    def get(self, request, unique_identifier):
+        if not UserInterest.objects.filter(
+            unique_identifier=unique_identifier
+        ).exists():
+            return Response({}, status=status.HTTP_404_NOT_FOUND)
+
+        user_interest = UserInterest.objects.get(unique_identifier=unique_identifier)
+        user_interest.waitlist = False
+        user = user_interest.user
+        if user.has_usable_password():
+            return Response({}, status=status.HTTP_410_GONE)
+        user_interest.save()
+        return Response({"email": user.email}, status=status.HTTP_200_OK)
+
+
+class InvitedUserSetPassword(APIView):
+    def post(self, request, unique_identifier):
+        if not UserInterest.objects.filter(
+            unique_identifier=unique_identifier
+        ).exists():
+            return Response({}, status=status.HTTP_404_NOT_FOUND)
+
+        user_interest = UserInterest.objects.get(unique_identifier=unique_identifier)
+        user = user_interest.user
+        if user.has_usable_password():
+            return Response({}, status=status.HTTP_410_GONE)
+        data = request.data
+        password = data.get("password", None)
```

Comment:
nit: this can be reduced to a one liner as the `data` variables isn't used anywhere else
```suggestion
        password = request.data.get("password", None)
```